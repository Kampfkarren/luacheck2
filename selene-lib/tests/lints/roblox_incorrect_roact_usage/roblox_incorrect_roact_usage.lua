Roact.createElement("Frame", {
    ThisPropertyDoesntExist = true,
    Size = UDim2.new(1, 0, 1, 0),

    [Roact.Event.InputBegan] = function()
    end,

    [Roact.Event.ThisEventDoesntExist] = function()
    end,
})

local e = Roact.createElement

e("Frame", {
    Size = UDim2.new(1, 0, 1, 0),
    ThisPropertyDoesntExist = true,
})

e("ThisDoesntExist", {})

e(Components.FooComponent, {
    Foo = 1,
})

call("foo", {})

e("ThisDoesntExist")

e(Components.FooComponent, {
    Name = "Can be passed",
})

Roact.createElement(Components.FooComponent, {
    Name = "Can be passed",
})

e("Frame", {
    Name = "Should not be passed",
})

e("Frame", {
    Name = "Should.not.be.passed",
})

e("Frame", {
    Name = "ShouldNotBePassed",
})

Roact.createElement("Frame", {
    Name = a,
})
