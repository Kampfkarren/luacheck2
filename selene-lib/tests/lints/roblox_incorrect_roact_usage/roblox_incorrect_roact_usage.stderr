error[roblox_incorrect_roact_usage]: `ThisPropertyDoesntExist` is not a property of `Frame`
  ┌─ roblox_incorrect_roact_usage.lua:2:5
  │
2 │     ThisPropertyDoesntExist = true,
  │     ^^^^^^^^^^^^^^^^^^^^^^^

error[roblox_incorrect_roact_usage]: `ThisEventDoesntExist` is not a valid event for `Frame`
  ┌─ roblox_incorrect_roact_usage.lua:8:5
  │
8 │     [Roact.Event.ThisEventDoesntExist] = function()
  │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[roblox_incorrect_roact_usage]: `ThisPropertyDoesntExist` is not a property of `Frame`
   ┌─ roblox_incorrect_roact_usage.lua:16:5
   │
16 │     ThisPropertyDoesntExist = true,
   │     ^^^^^^^^^^^^^^^^^^^^^^^

error[roblox_incorrect_roact_usage]: `ThisDoesntExist` is not a valid class
   ┌─ roblox_incorrect_roact_usage.lua:19:3
   │
19 │ e("ThisDoesntExist", {})
   │   ^^^^^^^^^^^^^^^^^

error[roblox_incorrect_roact_usage]: `ThisDoesntExist` is not a valid class
   ┌─ roblox_incorrect_roact_usage.lua:27:3
   │
27 │ e("ThisDoesntExist")
   │   ^^^^^^^^^^^^^^^^^

error[roblox_incorrect_roact_usage]: `Name` is assigned through the element's key for host components
   ┌─ roblox_incorrect_roact_usage.lua:30:5
   │
30 │     Name = "Should not be passed",
   │     ^^^^

